### PR TITLE
DOC: Add README.md

### PR DESCRIPTION
**Pull request by aresnow1**
## What do these changes do?

Add README.md.

## Related issue number

Fixes #xxxx

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,141 @@
...
+        for args, kwargs in zip(args_list, kwargs_list):
+            a, b = self.batch_method.bind(*args, **kwargs)
+            # process the request
+            results.append(result)
+        return results  # return a list of results
+```
+In a batch method, users can define how to more efficiently process a batch of requests.
+
+#### Call batch methods
+Calling batch methods is easy. You can use `<method_name>.delay` to make a batched call and `use <method_name>.batch` to send them:
```
Comment (UranusSeven) [inline README.md:133]:
```suggestion
Calling batch methods is easy. You can use `<method_name>.delay` to make a batched call and use `<method_name>.batch` to send them:
```

**Review by Hank0626 — DISMISSED** [review]

File: README.md [hunk 2]
```diff
@@ -3,3 +3,141 @@
 </div>
 
 # Python actor framework for heterogeneous computing.
+[![PyPI Latest Release](https://img.shields.io/pypi/v/xoscar.svg?style=for-the-badge)](https://pypi.org/project/xoscar/)
+[![Coverage](https://img.shields.io/codecov/c/github/xprobe-inc/xoscar?style=for-the-badge)](https://codecov.io/gh/xprobe-inc/xoscar)
+[![Build Status](https://img.shields.io/github/actions/workflow/status/xprobe-inc/xoscar/python.yaml?branch=main&style=for-the-badge&label=GITHUB%20ACTIONS&logo=github)](https://actions-badge.atrox.dev/xprobe-inc/xoscar/goto?ref=main)
+[![License](https://img.shields.io/pypi/l/xoscar.svg?style=for-the-badge)](https://github.com/xprobe-inc/xoscar/blob/main/LICENSE)
+
+## What is actor
+Writing parallel and distributed programs is often very challenging and requires a lot of time to deal with concurrency
```
Comment (qianduoduo0904) [inline README.md:12]:
```suggestion
Writing parallel and distributed programs is often challenging and requires a lot of time to deal with concurrency
```

File: README.md [hunk 3]
```diff
@@ -3,3 +3,141 @@
...
+- Batch method: Xoscar provides a batch interface to significantly improve call efficiency when an actor interface is 
+invoked a large number of times.
+- Advanced debugging support: Xoscar can detect potential issues like deadlocks, long-running calls, and performance
+bottlenecks that would otherwise be nearly impossible to troubleshoot in a heterogeneous environment.
+- Automated recovery: If an actor fails for any reason, Xoscar will automatically restart it if you want. It can monitor 
+actors and restart them upon failure, enabling fault-tolerant systems.
+
+## Overview 
+![architecture.png](doc/source/architecture.png)
+Xoscar allows you to create multiple actor pools on each worker node, typically binding an actor pool to a process or
```
Comment (qianduoduo0904) [inline README.md:37]: Maybe to a CPU or GPU?

File: README.md [hunk 4]
```diff
@@ -3,3 +3,141 @@
...
+
+When actors communicate, Xoscar will choose the optimal communication mechanism based on which pools the actors 
+belong to. This allows Xoscar to optimize communication in heterogeneous environments with multiple processing 
+units and accelerators.
+
+## Where to get it
+The source code is currently hosted on GitHub at: https://github.com/xprobe-inc/xoscar
+
+Binary installers for the latest released version are available at the [Python
+Package Index (PyPI)](https://pypi.org/project/xoscar).
```
Comment (qianduoduo0904) [inline README.md:49]: Lack of `pip install xoscar`?

File: README.md [hunk 5]
```diff
@@ -3,3 +3,141 @@
...
+actor_ref = await xo.actor_ref(worker_addr, actor_id)
+```
+
+#### Calling actor method
+```python
+# send
+await actor_ref.method_a.send(1, 2, a=1, b=2)
+# equivalent to actor_ref.method_a.send
+await actor_ref.method_a(1, 2, a=1, b=2)
+# tell
```
Comment (qianduoduo0904) [inline README.md:99]: Explain what's tell.

**Review by qianduoduo0904 — DISMISSED** [review]
LGTM

File: README.md [hunk 6]
```diff
@@ -3,3 +3,146 @@
...
+actor_ref = await xo.create_actor(
+    MyActor, 1, 2, a=1, b=2,
+    address='<ip>:<port>', uid='UniqueActorName')
+```
+
+#### Getting actor reference
+```python
+import xoscar as xo
+
+actor_ref = await xo.actor_ref(worker_addr, actor_id)
```
Comment (UranusSeven) [inline README.md:95]:
```suggestion
actor_ref = await xo.actor_ref(address, actor_id)
```

File: README.md [hunk 7]
```diff
@@ -3,3 +3,146 @@
...
+#### Creating actors
+```python
+import xoscar as xo
+
+actor_ref = await xo.create_actor(
+    MyActor, 1, 2, a=1, b=2,
+    address='<ip>:<port>', uid='UniqueActorName')
+```
+
+#### Getting actor reference
```
Comment (UranusSeven) [inline README.md:91]:
```suggestion
#### Get an actor reference
```

File: README.md [hunk 8]
```diff
@@ -3,3 +3,146 @@
...
+```python
+import xoscar as xo
+
+actor_ref = await xo.actor_ref(address, actor_id)
+```
+
+#### Calling actor method
```
Comment (UranusSeven) [inline README.md:98]:
```suggestion
#### Invoke a method
```

File: README.md [hunk 9]
```diff
@@ -3,3 +3,146 @@
...
+        results = []
+        for args, kwargs in zip(args_list, kwargs_list):
+            a, b = self.batch_method.bind(*args, **kwargs)
+            # process the request
+            results.append(result)
+        return results  # return a list of results
+```
+In a batch method, users can define how to more efficiently process a batch of requests.
+
+#### Call batch methods
```
Comment (UranusSeven) [inline README.md:137]:
```suggestion
#### Invoke a batch method
```

**Review by UranusSeven — APPROVED** [review]
LGTM.

**Review by UranusSeven — APPROVED** [review]
LGTM.

**Review by qianduoduo0904 — APPROVED** [review]
LGTM

**Pull request merged**